pfs_middleware/pfs_middleware/middleware.py

```diff
@@ -709,8 +709,25 @@ def __call__(self, req):
                     if denial_response:
                         return denial_response
 
-                # Authorization succeeded; dispatch to a helper method
+                # Authorization succeeded
                 method = req.method
+
+                # Check whether we ought to bypass. Note that swift_owner
+                # won't be set until we call authorize
+                if config_true_value(req.headers.get('X-Bypass-ProxyFS')) and \
+                        req.environ.get('swift_owner'):
+                    if method not in ('GET', 'HEAD'):
```

Comment:
While the point about how trivial it is to muck up the underlying file system metadata, I nevertheless think we should allow PUT here as well. Perhaps we could restrict it to not allow PUTs in the Checkpoint Container... but I'm not sure pfs_middleware knows what that is. Perhaps it should...